test/test_manifest.py

```diff
@@ -344,3 +345,100 @@ def test_manifest_target_arch_smoke(build_container, tc):
     # table be beside this there is relatively little that is different
     assert manifest["version"] == "2"
     assert manifest["pipelines"][0]["name"] == "build"
+
+
+def find_fstab_stage_from(manifest_str):
+    manifest = json.loads(manifest_str)
+    for pipeline in manifest["pipelines"]:
+        # the fstab stage in cross-arch manifests is in the "ostree-deployment" pipeline
+        if pipeline["name"] in ("image", "ostree-deployment"):
+            for st in pipeline["stages"]:
+                if st["type"] == "org.osbuild.fstab":
+                    return st
+    raise ValueError(f"cannot find fstab stage in manifest:\n{manifest_str}")
+
+
+@pytest.mark.parametrize("fscustomizations,rootfs", [
+    ({"/var/data": "2 GiB", "/var/stuff": "10 GiB"}, "xfs"),
+    ({"/var/data": "2 GiB", "/var/stuff": "10 GiB"}, "ext4"),
+    ({"/": "2 GiB", "/boot": "1 GiB"}, "ext4"),
+    ({"/": "2 GiB", "/boot": "1 GiB", "/var/data": "42 GiB"}, "ext4"),
+    ({"/": "2 GiB"}, "btrfs"),
+])
+def test_manifest_fs_customizations(tmp_path, build_container, fscustomizations, rootfs):
```

Comment:
I wonder if we can cover many of these cases in unit tests maybe? I think it's important to cover at least one integration test here but testing many cases at this level is relatively costly (the test went from 2min to 7min on my system) so maybe move some into go-unit tests and just cover one scenario (or two?) here.

Reply:
I wanted an integration test to run the paths starting from the `rootfs` command line option all the way down, and having a real container in the mix with its own rootfs type config should be overridden.  Code paths under that should be mostly covered by unit tests already, but this will catch regressions or changes with how the rootfs choice interacts with the customizations, which surfaces issues like the rootfs being empty when creating cross-arch manifests.

I think a lot of this can be sped up if we had a way to mock the base container, but that's rather difficult with how much we depend on information from the base container read through actually running or mounting it.  If we could do that, our manifest integration tests would take a few seconds.

Reply:
Thank you! Totally fine to do it this way, just wanted to have the conversation :) I will mark this as resolved.

Reply:
FTR, I compared integration test timing between this branch and `main` (for manifest tests).  Running
```
pytest -k test_manifest
```
I get, on `main`:
```
slowest 10 durations 
344.91s call     test/test_manifest.py::test_manifest_disksize[quay.io/centos-bootc/centos-bootc:stream9]
338.54s call     test/test_manifest.py::test_manifest_disksize[quay.io/fedora/fedora-bootc:40,btrfs]
78.21s setup    test/test_manifest.py::test_manifest_smoke[quay.io/centos-bootc/centos-bootc:stream9]
20.21s call     test/test_manifest.py::test_manifest_anaconda_module_customizations[quay.io/fedora/fedora-bootc:40,anaconda-iso,btrfs]
17.64s call     test/test_manifest.py::test_manifest_installer_customizations
16.61s call     test/test_manifest.py::test_iso_manifest_smoke[quay.io/fedora/fedora-bootc:40,anaconda-iso,btrfs]
16.48s call     test/test_manifest.py::test_iso_manifest_smoke[quay.io/centos-bootc/centos-bootc:stream9,anaconda-iso]
16.20s call     test/test_manifest.py::test_manifest_anaconda_module_customizations[quay.io/centos-bootc/centos-bootc:stream9,anaconda-iso]
16.19s call     test/test_manifest.py::test_manifest_target_arch_smoke[quay.io/centos-bootc/centos-bootc:stream9,arm64]
11.82s call     test/test_manifest.py::test_manifest_target_arch_smoke[quay.io/fedora/fedora-bootc:41,btrfs,ppc64le]
22 passed, 44 deselected in 960.91s (0:16:00) 
```
and on the PR branch:
```
slowest 10 durations 

339.93s call     test/test_manifest.py::test_manifest_disksize[quay.io/fedora/fedora-bootc:40,btrfs]
316.35s call     test/test_manifest.py::test_manifest_disksize[quay.io/centos-bootc/centos-bootc:stream9]
80.15s setup    test/test_manifest.py::test_manifest_smoke[quay.io/centos-bootc/centos-bootc:stream9]
18.54s call     test/test_manifest.py::test_iso_manifest_smoke[quay.io/fedora/fedora-bootc:40,anaconda-iso,btrfs]
17.63s call     test/test_manifest.py::test_manifest_anaconda_module_customizations[quay.io/fedora/fedora-bootc:40,anaconda-iso,btrfs]
17.54s call     test/test_manifest.py::test_manifest_installer_customizations
17.09s call     test/test_manifest.py::test_manifest_anaconda_module_customizations[quay.io/centos-bootc/centos-bootc:stream9,anaconda-iso]
16.52s call     test/test_manifest.py::test_manifest_target_arch_smoke[quay.io/centos-bootc/centos-bootc:stream9,arm64]
15.43s call     test/test_manifest.py::test_iso_manifest_smoke[quay.io/centos-bootc/centos-bootc:stream9,anaconda-iso]
12.23s call     test/test_manifest.py::test_manifest_target_arch_smoke[quay.io/fedora/fedora-bootc:41,btrfs,s390x]

22 passed, 44 deselected in 931.73s (0:15:31) 
```

I generally agree, we could reduce the number of these or make them faster by mocking some things and moving them to Go tests, but I think for now the impact of these new ones isn't high.

Reply:
Thanks, yeah, that is really a small difference, very nice!